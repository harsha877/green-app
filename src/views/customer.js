import React, { useState } from "react";
import { View, Text, Button, ScrollView, SafeAreaView, StyleSheet } from 'react-native';
import KeyboardAvoidingView from "react-native/Libraries/Components/Keyboard/KeyboardAvoidingView";
import CustomerHomeScreen from "./customerHomeScreen";
import CustomerLogin from "./customerLogin";
import QuizInput from "./QuizInput";
export default function Customer({ navigation }) {
    const [user, setUser] = useState("");

    const onSubmitHandler = (data) => {
        setUser(data);
    };

    return (
        <KeyboardAvoidingView style={styles.container}>
            {user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <CustomerHomeScreen onSubmitHandler = {onSubmitHandler} navigation = {navigation} />}
        </KeyboardAvoidingView>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    }
});