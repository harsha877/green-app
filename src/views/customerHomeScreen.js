import React, { useState } from "react";
import { View, Text, Button, ScrollView, SafeAreaView, StyleSheet } from 'react-native';
import KeyboardAvoidingView from "react-native/Libraries/Components/Keyboard/KeyboardAvoidingView";

export default function CustomerHomeScreen({ navigation, onSubmitHandler }) {



    const handleCreateQuiz = () => {
        navigation.navigate('customer question');
    }

    const handleViewQuizzes = () =>{
        console.log("view quizzes");
    }

    return (
        <View style={styles.container}>
            <Text>Customer Homepage</Text>
            <Button  onPress={handleCreateQuiz} title='Create Quiz' />
            <Button  onPress={handleViewQuizzes} title='View Quizzes' />
            <Button  onPress={() => onSubmitHandler("")} title='Log Out' />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    }
});


//{user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <QuizInput setUser={setUser} navigation = {navigation} />}