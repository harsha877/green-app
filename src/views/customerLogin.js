import * as React from 'react';
import { View, Text } from 'react-native';
// import bottomNav from './styles/bottomNav';

export default function CustomerLogin({ navigation }) {
    return (
        const [email, setEmail] = useState("");
         const [password, setPassword] = useState("");
 
   function handleLogin(e)
   {
      e.preventDefault();
      console.log(email);
      console.log(password);
   }
  return (
    <View style={styles.container}>

      <Image source={require('../assets/avatar.png')}/>
        <TextInput
        style={styles.TextInput}
          placeholder="Email"
          value={email}
          onChangeText={(email) => setEmail(email)}
        />

        <TextInput
        style={styles.TextInput}
          placeholder="Password"
          secureTextEntry={true}
          value={password}
          onChangeText={(password) => setPassword(password)}
        />
 
        <Button style={styles.Button} onPress={handleLogin} title='LOGIN'/>
    </View>
  );
}
 
const styles = StyleSheet.create({
  container: {
    display:"flex",
    flexDirection:"column",
    justifyContent:"center",
    alignItems:"center",
    marginTop:200
  },
 
  TextInput:  {
        height: 40,
        margin:15,
        width:250,
        borderWidth: 1,
        padding: 10,
      },

 
  Button:
  {
        height: 40,
        borderWidth: 1,
        marginTop:10,
        padding: 10,
  }
});
