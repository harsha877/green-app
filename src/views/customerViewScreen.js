import React, { useState } from "react";
import { View, Text, TextInput, ScrollView, StyleSheet } from 'react-native';
import { Button } from "react-native-web";
import CustumeButton from "../components/button";
import CustumeTable from "../components/table";
import { GREEN_DB_COLLECTION_QUIZES } from "../constant/constants";
import { searchQuizesForCustomer, writeCollection } from "../firebase/config";


export default function CustomerViewScreen({ navigation, route }) {

    const [user, setUser] = useState(route.params.user);
    const [quizzes, setQuizzes] = useState();
    const [quizFlag, setQuizFlag] = useState(true);
    const [optionsFlag, setOptionsFlag] = useState(false);
    const [editQuestionFlag, seteditQuestionFlag] = useState(false);
    const [selectedIndex, setSelectedIndex] = useState();
    const [questionIndex, setQuestionIndex] = useState();
    const [quiz, setQuiz] = useState();
    const [textBoxSize, setTextBoxSize] = useState(100);

    const handleViewQuizzes = (index) => {
        setSelectedIndex(index);
        setQuiz(quizzes[index])
        console.log("selected " + quizzes[index].quizName);
        setOptionsFlag(true);
        setQuizFlag(false);
        showQuestions();
    }

    const showQuestions = () => {
        if (optionsFlag && selectedIndex != undefined && quiz != undefined) {
            //console.log(quiz.questions.map(data => data));
            return (
                quiz.questions.map((data, index) =>
                    <Text style={styles.questions} onPress={() => setQuestionIndex(index)} key={index}>
                        {data.question + '\n'}
                    </Text>)
            );
        } else {
            console.log("view questions validations failed");
        }
    }


    const getData = () => {
        searchQuizesForCustomer(route.params.user.username).then(data => {
            setQuizzes(data);
            viewList();
            showQuestions();
        })
    }


    const viewList = () => {

        if (quizFlag && quizzes != undefined) {
            console.log("showing List of quizes")
            return (quizzes.map(
                (data, index) => {
                    //console.log(a);
                    return (
                        <CustumeButton key={index}
                            onPressHandler={() => handleViewQuizzes(index)}
                            name={data.quizName}
                            type='button' />
                    )
                }));
        }

    };

    const viewQuestions = () => {

        if (questionIndex != undefined) {
            return (
                <View style={styles.questionsEdit}>
                    <Text>Question: </Text>
                    <TextInput 
                        style={{ ...styles.textBoxes, height: textBoxSize }}
                        value={quiz.questions[questionIndex].question} 
                        onChangeText = {(question) => updatequestion("qestion",question)} 
                        multiline
                        onContentSizeChange={(e) => setTextBoxSize(e.nativeEvent.contentSize.height)}
                        />
                    <Text>Option 1: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option1}
                        onChangeText = {(question) => updatequestion("option1",question)} />
                    <Text>Option 2: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option2}
                        onChangeText = {(question) => updatequestion("option2",question)} />
                    <Text>Option 3: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option3}
                        onChangeText = {(question) => updatequestion("option3",question)}/>
                    <Text>Option 4: </Text>
                    <TextInput style= {styles.textBoxes} 
                        value={quiz.questions[questionIndex].option4}
                        onChangeText = {(question) => updatequestion("option4",question)}/>
                    <CustumeButton name="Save" onPressHandler={saveUpdatedData}/>
                </View>

            )
        }
    }

    const updatequestion = (topic ,text) => {
        let temp = {...quiz};
        if (topic === "qestion"){
            temp.questions[questionIndex].question = text;
        }else if (topic === "option1"){
            temp.questions[questionIndex].option1 = text;
        }else if (topic === "option2"){
            temp.questions[questionIndex].option2 = text;
        }else if (topic === "option3"){
            temp.questions[questionIndex].option3 = text;
        }else if (topic === "option4"){
            temp.questions[questionIndex].option4 = text;
        }
        setQuiz(temp);
    }

    const saveUpdatedData = () => {

        console.log(quiz.quizName+ " : data update");
        writeCollection(quiz, quiz.quizName, GREEN_DB_COLLECTION_QUIZES);
        alert("Changes Saved!");
        //to do need to uncomment according to requirment
        //navigation.navigate('customer login', {user});
    }

    const viewrefresh = () => {
        if (quizFlag) {
            return (<CustumeButton name="refresh" onPressHandler={getData} />)
        }
    }

    return (
        <ScrollView >
            <Text>Customer View page</Text>

            <View style={styles.list}>
                <CustumeButton name="refresh" onPressHandler={getData} type="button" />
                <Text>{viewList()}</Text>
            </View>
            <View style={styles.list}>{showQuestions()}</View>
            {viewQuestions()}

        </ScrollView>
    )
}


const styles = StyleSheet.create({
    container: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        marginTop: 30
    },
    list: {
        justifyContent: 'space-between',
        flexDirection: 'column',
        justifyContent: "center",
        alignItems: "center",
    },
    questions: {
        //alignItems: "center",
        backgroundColor: '#1992d4',
        padding: 10,
        margin: 10,
        borderRadius: 10,
    },
    questionsEdit: {
        //alignItems: "center",
        backgroundColor: '#1992d4',
        padding: 10,
        margin: 10,
        borderRadius: 10,
    },
    textBoxes: {
        margin: 15,
        //width: 250,
        borderWidth: 1,
        padding: 10,
        borderRadius: 10,
    }
});


//{user == "" ? <CustomerLogin onSubmit={onSubmitHandler} /> : <QuizInput setUser={setUser} navigation = {navigation} />}