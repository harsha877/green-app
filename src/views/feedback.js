import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet,ScrollView,Dimensions,TouchableOpacity,SafeAreaView,Image, TextInput,Button, Alert
 } from 'react-native';
import Constants from 'expo-constants';
import TypeWriter from "react-native-typewriter";
import RadioGroup, {RadioButtonProps} from 'react-native-radio-buttons-group';


const radioButtonsData = [{
    id: '1', // acts as primary key, should be unique and non-empty string
    label: 'Yes',
    value: 'Y'
    }, {
    id: '2',
    label: 'No',
    value: 'N'
    }]

export default function FeedBack({ route, navigation }) {
    const [text, onChangeText] = React.useState("");
    const [defaultRating,setdefaultRating] = useState(0)
    const [maxRating,setmaxRating]= useState([1,2,3,4,5])
    const [radioButtons, setRadioButtons] = useState(radioButtonsData)
            
    

    const RadioButtn = () => {
      
       function onPressRadioButton(radioButtonsArray) {
        setRadioButtons(radioButtonsArray);
      }
      return (
        <RadioGroup 
            radioButtons={radioButtons} 
            onPress={onPressRadioButton} 
        />
    );

    }

  const UseTextInput = (props) => {
  return (
    <TextInput
      {...props} // Inherit any props passed to it; e.g., multiline, numberOfLines below
      editable
      maxLength={40}
    />
  );
}

const UseTextInputMultiline = () => {
  const [value, onChangeText] = React.useState('');

  
  return (
    <View
      style={{
        backgroundColor: value,
        borderBottomColor: '#000000',
        borderWidth: 1,
      }}>
      <UseTextInput
        multiline
        numberOfLines={4}
        onChangeText={text => onChangeText(text)}
        value={value}
        style={{padding: 5}}
      />
    </View>
  );
}



    const CustomRatingBar = () => {
        return(
            <View style={styles.customRatingBarStyle}>
            {
              maxRating.map((item,key) => {
                return(
                  <TouchableOpacity
                  activeOpacity={0.7}
                  key={item}
                  onPress={() => setdefaultRating(item)}

                  >
                  <Image
                  style={styles.starImgStyle}
                  source={
                    item <= defaultRating
                    ? require("../../assets/star_filled.png")
                    : require("../../assets/star_corner.png")
                  }
                  />
                  </TouchableOpacity>
                )
              }

              )
            }
            </View>

        )
    }

    const data= [
      {
        "question": "Q1. How would you rate your overall experience?",
        "id": 1
      },
      {
        "question": "Q2. Was our App Useful?",
        "id":2
      },
      {
        "question": "Q3. Comments ",   
        "id":3 
      }
    ]

    const SubmitFeedBack = () =>{
        Alert.alert(
          "Feedback",
          "Your Feedback Submitted",
          [
        {
          text: "OK",
          onPress: () => navigation.goBack()
        }
      ]
        );

    }


    const FeedBackList = () => {
        return(
             <View style={styles.feedbackQuestionStyle}>
               {
                data.map((item,key) => {
                  return(
                    <TouchableOpacity key={key}>
                     <Text style={styles.Question}>
                       {item.question}
                     </Text>
                     {
                      item.id == 1
                     ?
                      <View>
                      <CustomRatingBar />  
                       <Text style={styles.textStyle}>
                        {defaultRating + '/' + maxRating.length}
                      </Text>
                      </View>
                     : 
                     item.id == 2
                     ?
                            <RadioButtn style={styles.radiostyle}/>
                     :
                            <UseTextInputMultiline/>                     
                     }
                  </TouchableOpacity>
                 
                  )
                   }
                )
        }
         </View>

     )
}

return (
  <SafeAreaView style={styles.container}>
  <ScrollView >
    <View style={styles.container}>
      <Text style={styles.text}>
              Rate App
      </Text>
      <FeedBackList/>
      
    </View>
    <View style={styles.subbutton}>
    <Button color={"#063f5c"}  title="Submit" onPress={SubmitFeedBack}  />
    </View>
  </ScrollView>
  </SafeAreaView>  
);

}






const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center",
		backgroundColor: "#ecf0f1",
		padding: 2,
	},
	paragraph: {
		borderColor: "black",
		borderWidth: 1,
		margin: 30,
		marginBottom: 3,
		borderRadius: 25,
		height: 55,
		backgroundColor: "#003f69",
		fontSize: 18,
		textAlign: "center",
		paddingHorizontal: 30,
	},
  text: {
		color: "#000000",
		fontSize: 20,
		fontWeight: "600",
    marginTop: 20
	},
	typeWriter: {
		color: "#2471A3",
		fontSize: 25,
		margin: 15,
		fontWeight: "500",
		textAlign: "center",
	},
	questionNo: {
		color: "#2471A3",
		fontSize: 18,
		margin: 5,
	},
	Question: {
		fontSize: 20,
		margin: 5,
		fontWeight: "500",
	},
	top: {},
	bottom: {
		position: "absolute",
		width: Dimensions.get("screen").width,
		bottom: 10,
	},
	box: {
		backgroundColor: "#2471A3",
		height: 80,
		marginTop: 110,
	},
	bottomWavy: {
		position: "absolute",
		bottom: 20,
	},
  customRatingBarStyle: {
    justifyContent: 'center',
    flexDirection: 'row',
    marginTop: 30
  },
  feedbackQuestionStyle: {
    justifyContent: 'center',
    flexDirection: 'column',
  },
  starImgStyle:{
    width: 40,
    height: 40,
    resizeMode: 'cover'
  },
  radiostyle:{
    justifyContent: 'center',
    flexDirection: 'column'
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  subbutton: {
   
    fontsize: 30,
    margin:10,
    padding: 10,
    marginTop: 20,
		borderRadius: 50
   },
  textStyle: {
    textAlign: 'center',
    fontSize: 23,
    marginTop: 20
  }
})
