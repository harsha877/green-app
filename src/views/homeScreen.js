import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView} from 'react-native';
import TypeWriter from 'react-native-typewriter';
import Questions from '../json/questions.json';
import Option from '../components/options.js';
import CustumeButton from '../components/button';

export default function HomeScreen({route, navigation}) {
  const {index} = route.params;
  return (
    <ScrollView>
      <TypeWriter style={styles.typeWriter} typing={1}>
        Water Footprint Calculator
      </TypeWriter>
      <View style={styles.container}>
        <Text style={styles.questionNo}>
          {index + 1}/{Questions.questions.length}
        </Text>
        <Text style={styles.Question}>
          {Questions.questions[index].question}
        </Text>
        {Questions.questions[index].answers.map(user => {
          let keys = Object.keys(user);
          return (
            <Text style={styles.paragraph}>
              {keys.map(key => {
                return (
                  <Option
                    value={`${key}`}
                    optionIdx={`${user[key]}`}
                    navigation={navigation}
                    qnIndex={index}
                  />
                );
              })}
            </Text>
          );
        })}
      </View>
    </ScrollView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#ecf0f1',
    padding: 2,
  },
  paragraph: {
    borderColor: 'black',
    borderWidth: 1,
    margin: 20,
    marginBottom: 3,
    borderRadius: 20,
    // height: 65,
    padding: 20,
    backgroundColor: '#003f69',
    textAlign: 'center',
    color: '#000000',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    fontSize: 18,
  },
  typeWriter: {
    color: '#409cd0',
    fontSize: 25,
    margin: 15,
    fontWeight: '500',
    textAlign: 'center',
  },
  questionNo: {
    color: '#409cd0',
    fontSize: 18,
    margin: 5,
  },
  Question: {
    fontSize: 18,
    margin: 5,
  },
});
