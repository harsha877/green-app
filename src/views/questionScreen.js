import React, { useState } from "react"
import { View, Text, StyleSheet } from "react-native"
import CustumeButton from "../components/button"
import {
	USER_CUSTOM_QUIZ_HOMESCREEN,
	USER_CUSTOM_QUIZ_RESULTSCREEN,
} from "../constant/constants"

export default function QuestionScreen({ route, navigation }) {
	const [quiz, setQuiz] = useState(route.params.data[0])
	const [currentquestion, setCurrentQuestion] = useState(quiz.questions[0])
	const [currentquestionNumber, setCurrentquestionNumber] = useState(1)
	const [weights, setWeights] = useState([0])
	const [score, SetScore] = useState(0)
	const [color, setColor] = useState(["white", "white", "white", "white"])

	const ResultPage = () => {
		let sum = weights.reduce((pSum, element) => pSum + parseInt(element), 0)
		//console.log(sum);
		let result = (sum / (4 * quiz.length)) * 100
		console.log("user Scored: " + result + "%")
		navigation.navigate(USER_CUSTOM_QUIZ_RESULTSCREEN, { result })
	}

	const navigate = (flag) => {
		setColor(["white", "white", "white", "white"])
		if (flag == 1) {
			setCurrentQuestion(quiz.questions[currentquestionNumber])
			setCurrentquestionNumber(currentquestionNumber + 1)
			if (weights[currentquestionNumber] == undefined) {
				let tempWeights = weights
				tempWeights.push(0)
				setWeights(tempWeights)
			}
			console.log(weights)
		} else {
			setCurrentQuestion(quiz.questions[currentquestionNumber - 2])
			setCurrentquestionNumber(currentquestionNumber - 1)
			console.log(weights)
		}
	}

	const changeColor = (weight) => {
		if (weight == 1) {
			setColor(["green", "white", "white", "white"])
		} else if (weight == 2) {
			setColor(["white", "green", "white", "white"])
		} else if (weight == 3) {
			setColor(["white", "white", "green", "white"])
		} else if (weight == 4) {
			setColor(["white", "white", "white", "green"])
		}
	}

	const optionWeight = (weight) => {
		changeColor(weight)
		let tempWeights = weights
		tempWeights[currentquestionNumber - 1] =
			currentquestion.weight[weight - 1]
		setWeights(tempWeights)
		console.log(weights)
	}

	const previousButtonView = () => {
		if (currentquestionNumber != 1) {
			return (
				<CustumeButton
					name="Previous"
					onPressHandler={() => navigate(-1)}
				/>
			)
		}
	}
	const nextButtonView = () => {
		if (currentquestionNumber != quiz.length) {
			return (
				<CustumeButton name="Next" onPressHandler={() => navigate(1)} />
			)
		}
	}
	const subitButtonView = () => {
		if (currentquestionNumber == quiz.length) {
			return <CustumeButton name="Submit" onPressHandler={ResultPage} />
		}
	}

	//console.log(quiz);

	return (
		<View>
			<View>
				<Text style={styles.questionNo}>
					{currentquestionNumber}/{quiz.length}
				</Text>
				<Text style={styles.Question}>{currentquestion.question}</Text>
			</View>
			<View>
				<View style={{ backgroundColor: color[0], ...styles.options }}>
					<CustumeButton
						name={currentquestion.option1}
						onPressHandler={() => optionWeight(1)}
					/>
				</View>
				<View style={{ backgroundColor: color[1], ...styles.options }}>
					<CustumeButton
						name={currentquestion.option2}
						onPressHandler={() => optionWeight(2)}
					/>
				</View>
				<View style={{ backgroundColor: color[2], ...styles.options }}>
					<CustumeButton
						name={currentquestion.option3}
						onPressHandler={() => optionWeight(3)}
					/>
				</View>
				<View style={{ backgroundColor: color[3], ...styles.options }}>
					<CustumeButton
						name={currentquestion.option4}
						onPressHandler={() => optionWeight(4)}
					/>
				</View>
			</View>

			<View style={styles.navigation}>
				{previousButtonView()}
				{nextButtonView()}
				{subitButtonView()}
			</View>
		</View>
	)
}

const styles = StyleSheet.create({
	options: {
		margin: 0,
		padding: 0,
		margin: 30,
		marginBottom: 3,
		borderRadius: 25,
		backgroundColor: "#42bff5",
		fontSize: 18,
		textAlign: "center",
	},
	navigation: {
		justifyContent: "space-between",
		flexDirection: "row",
		alignItems: "center",
		justifyContent: "center",
		marginTop: 30,
	},
	questionNo: {
		color: "#003f69",
		fontSize: 18,
		margin: 5,
	},
	Question: {
		fontSize: 20,
		margin: 5,
		fontWeight: "500",
		textAlign: "center",
	},
})
