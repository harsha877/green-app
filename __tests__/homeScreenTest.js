import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import HomeScreen from '../src/views/homeScreen';
import { useStateValue } from "../src/views/stateProvider"
import { TestWatcher } from 'jest';

test("test", () => {})
/*
var routes = { params: 0}
test("Options Component Loading", () => {
    Questions = jest.mock('src/json/questions.json', () => (
        {
            "questions": [
                {
                    "question": "How long is the average shower in your household?",
                    "type": "HouseHold",
                    "answers": [
                        { "Under 5 min": 200 },
                        { "5-10 min": 300 },
                        { "11-15 min": 400 },
                        { "Over 15 min": 500 }
                    ]
                },]
        }
        
    ), { virtual: true })

    render(<HomeScreen route= {routes}  navigation ={NavigationContainer} />)

})

/*
test("Option component snapshot", () => {
    let component = render(<Options prop={arr} />).toJSON();
    expect(component).toMatchSnapshot();
})


describe("Testing Option component", () => {
    let component;
    beforeEach(() => {
        component = render(<Options prop={arr} />);

    })

    test("Testing Styles", () => {
        const props = component.toJSON();
        //console.log(props);
        let Styles = props.children[0].props.style;
        //console.log(Styles);
        expect(Styles).toStrictEqual({
            opacity: 1,
        });
    })

    test("Testing Styles text", () => {
        const props = component.toJSON();
        //console.log(props);
        let Styles = props.children[0].children[0].props.style;
        console.log(Styles);
        expect(Styles).toStrictEqual({
            fontSize: 20,
            color: "white",
            textAlign: "center",
        });
    })

});*/