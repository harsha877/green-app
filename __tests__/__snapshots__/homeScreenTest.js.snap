// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot 1`] = `
<RCTScrollView>
  <View>
    <Image
      source={
        Object {
          "testUri": "../../../assets/app_logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 80,
          "marginTop": 30,
          "resizeMode": "contain",
          "width": 450,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#ecf0f1",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 20,
          "padding": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2471A3",
            "fontSize": 18,
            "margin": 5,
          }
        }
      >
        1
        /
        10
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "margin": 5,
          }
        }
      >
        How long is the average shower in your household?
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#003f69",
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 60,
            "margin": 30,
            "marginBottom": 3,
            "opacity": 1,
            "paddingHorizontal": 30,
            "textAlign": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 8,
                  "textAlign": "center",
                }
              }
            >
              Under 5 min
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#003f69",
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 60,
            "margin": 30,
            "marginBottom": 3,
            "opacity": 1,
            "paddingHorizontal": 30,
            "textAlign": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 8,
                  "textAlign": "center",
                }
              }
            >
              5-10 min
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#003f69",
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 60,
            "margin": 30,
            "marginBottom": 3,
            "opacity": 1,
            "paddingHorizontal": 30,
            "textAlign": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 8,
                  "textAlign": "center",
                }
              }
            >
              11-15 min
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#003f69",
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "fontSize": 18,
            "height": 60,
            "margin": 30,
            "marginBottom": 3,
            "opacity": 1,
            "paddingHorizontal": 30,
            "textAlign": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 8,
                  "textAlign": "center",
                }
              }
            >
              Over 15 min
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#2471A3",
            "height": 80,
            "marginTop": 130,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={200}
          bbWidth={750}
          focusable={false}
          height={200}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 200,
                "width": 750,
              },
            ]
          }
          vbHeight={320}
          vbWidth={1440}
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M0,192L60,170.7C120,149,240,107,360,112C480,117,600,171,720,197.3C840,224,960,224,1080,208C1200,192,1320,160,1380,144L1440,128L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
              fill={4280578467}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 10,
          "position": "absolute",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#2471A3",
            "height": 80,
            "marginTop": 130,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={200}
          bbWidth={750}
          focusable={false}
          height={200}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "bottom": 20,
                "position": "absolute",
              },
              Object {
                "flex": 0,
                "height": 200,
                "width": 750,
              },
            ]
          }
          vbHeight={320}
          vbWidth={1440}
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M0,64L40,96C80,128,160,192,240,202.7C320,213,400,171,480,149.3C560,128,640,128,720,154.7C800,181,880,235,960,218.7C1040,203,1120,117,1200,74.7C1280,32,1360,32,1400,32L1440,32L1440,320L1400,320C1360,320,1280,320,1200,320C1120,320,1040,320,960,320C880,320,800,320,720,320C640,320,560,320,480,320C400,320,320,320,240,320C160,320,80,320,40,320L0,320Z"
              fill={4280578467}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
